generators/entity-client/templates/react/src/main/webapp/app/entities/_entity.tsx

```diff
@@ -45,11 +54,22 @@ export interface I<%= entityReactName %>Props {
   <%_ } _%>
   match: any;
 }
+<%_ if (searchEngine === 'elasticsearch') { _%>
+
+export interface I<%= entityReactName %>State {
+  search: string;
+}
+<%_ } _%>
 
-export class <%= entityReactName %> extends React.Component<I<%= entityReactName %>Props, undefined> {
+export class <%= entityReactName %> extends React.Component<I<%= entityReactName %>Props, <% if (searchEngine === 'elasticsearch') { %>I<%= entityReactName %>State<% } else { %>undefined<% } %>> {
 
   constructor(props) {
     super(props);
+    <%_ if (searchEngine === 'elasticsearch') { _%>
```

Comment:
can you move this check to wrap the constructor as its not needed when there is no state